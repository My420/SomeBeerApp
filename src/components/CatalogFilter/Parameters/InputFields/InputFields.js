import React from 'react';
import styles from './InputFields.module.scss';
import Abv from './Abv/Abv';
import Ibu from './Ibu/Ibu';
import Ebc from './Ebc/Ebc';
import {
  ABV_LESS_PROP,
  ABV_MORE_PROP,
  IBU_LESS_PROP,
  IBU_MORE_PROP,
  EBC_LESS_PROP,
  EBC_MORE_PROP
} from '../../../../utils/constants';

const InputFields = ({ options, onPropertyChange }) => {
  const {
    [ABV_LESS_PROP]: abvLess,
    [ABV_MORE_PROP]: abvMore,
    [IBU_LESS_PROP]: ibuLess,
    [IBU_MORE_PROP]: ibuMore,
    [EBC_LESS_PROP]: ebcLess,
    [EBC_MORE_PROP]: ebcMore
  } = options;

  return (
    <div className={styles.wrapper}>
      <Abv
        lessValue={abvLess}
        moreValue={abvMore}
        onPropertyChange={onPropertyChange}
      />
      <Ibu
        lessValue={ibuLess}
        moreValue={ibuMore}
        onPropertyChange={onPropertyChange}
      />
      <Ebc
        lessValue={ebcLess}
        moreValue={ebcMore}
        onPropertyChange={onPropertyChange}
      />
    </div>
  );
};

export default InputFields;
