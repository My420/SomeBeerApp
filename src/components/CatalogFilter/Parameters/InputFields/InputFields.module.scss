@import '../../../../sass/variables.scss';

.wrapper {
  margin-top: 5px;
  @include border;
}
