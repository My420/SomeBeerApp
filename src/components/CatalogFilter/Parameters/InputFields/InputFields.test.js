import React from 'react';
import { shallow } from 'enzyme';
import InputFields from './InputFields';
import {
  ABV_LESS_PROP,
  ABV_MORE_PROP,
  IBU_LESS_PROP,
  IBU_MORE_PROP,
  EBC_LESS_PROP,
  EBC_MORE_PROP
} from '../../../../utils/constants';

describe('test <InputFields /> component', () => {
  const options = {
    [ABV_LESS_PROP]: '10',
    [ABV_MORE_PROP]: '20',
    [IBU_LESS_PROP]: '30',
    [IBU_MORE_PROP]: '40',
    [EBC_LESS_PROP]: '50',
    [EBC_MORE_PROP]: '60'
  };

  const wrapper = shallow(
    <InputFields options={options} onPropertyChange={() => {}} />
  );

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
