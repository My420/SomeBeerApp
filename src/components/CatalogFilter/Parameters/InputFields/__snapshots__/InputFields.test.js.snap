// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <InputFields /> component should match snapshot 1`] = `
<div
  className="wrapper"
>
  <Abv
    lessValue="10"
    moreValue="20"
    onPropertyChange={[Function]}
  />
  <Ibu
    lessValue="30"
    moreValue="40"
    onPropertyChange={[Function]}
  />
  <Ebc
    lessValue="50"
    moreValue="60"
    onPropertyChange={[Function]}
  />
</div>
`;
